lib/porky_lib/file_service.rb

```diff
@@ -59,18 +61,24 @@ def write(file, bucket_name, key_id, options = {})
     raise FileSizeTooLargeError, "File size is larger than maximum allowed size of #{max_file_size}" if file_size_invalid?(file)
 
     data = file_data(file)
-    file_key = generate_file_key(options)
-    tempfile = encrypt_file_contents(data, key_id, file_key, options)
+    write_helper(data, bucket_name, key_id, options)
+  end
+  deprecate :write, 'write_file or write_data', 2020, 1
 
-    begin
-      perform_upload(bucket_name, file_key, tempfile, options)
-    rescue Aws::Errors::ServiceError => e
-      raise FileServiceError, "Attempt to upload a file to S3 failed.\n#{e.message}"
-    end
+  def write_file(file, bucket_name, key_id, options = {})
+    raise FileServiceError, 'Invalid input. One or more input values is nil' if input_invalid?(file, bucket_name, key_id)
 
-    # Remove tempfile from disk
-    tempfile.unlink
-    file_key
+    data = File.read(file)
```

Comment:
... true I should do a `File.file?` before and raising an error if it doesn't exist

Reply:
File.res can throw
- `ArgumentError` 
  - `(wrong number of arguments (given 0, expected 1..4))` if it doesn't have the right number of arguments (which is impossible here since we are only using one)
  - `(path name contains null byte)` if it has null bytes in the path
- `TypeError (no implicit conversion of nil into String)` if the type of the param is nil (we are checking if the param is nil beforehand)
- `Errno::ENOENT (No such file or directory @ rb_sysopen - )` if the file doesn't exist (`File.file?` checks that)
- `Errno::EACCES` if you don't have permission for opening that file (I added a rescue for that, but not sure on how to test it)